### PR TITLE
Remediation File Added for Azure Front Door Non Classic BRS

### PR DESCRIPTION
**Pull request by jaiswalricha**
Script File to Configure WAF on Azure Front Door CDN Profile

### CODE REVIEW

**Review by VamseeInala — APPROVED** [review]

File: Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1 [hunk 1]
```diff
@@ -0,0 +1,795 @@
...
+
+# Control ID:
+    Azure_FrontDoor_CDNProfile_NetSec_Enable_WAF_Configuration
+
+# Display Name:
+   Front Door CDN should have Web Application Firewall configured
+
+# Prerequisites:
+    1. Contributor or higher privileges on the Front Door CDNs in a Subscription.
+    2. Must be connected to Azure with an authenticated account.
```
Comment (asimks) [inline Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1:13]: Can we say Must be connected to Azure with same authenticated account with which have required access over Front door CDNs.

File: Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1 [hunk 2]
```diff
@@ -0,0 +1,795 @@
...
+
+# Steps performed by the script:
+    To remediate:
+        1. Validate and install the modules required to run the script.
+        2. Get the list of all Front Door CDNs Endpoints in a Subscription that do not have WAF Policy Configured
+        3. Back up details of Front Door CDN Endpoint(s) that are to be remediated.
+        4. Configure WAF for all endpoints in the Front Door.
+
+    To roll back:
+        1. Validate and install the modules required to run the script.
```
Comment (asimks) [inline Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1:23]: Do we validate again in rollback?

File: Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1 [hunk 3]
```diff
@@ -0,0 +1,795 @@
...
+        2. Get the list of Front Door' Endpoint(s) in a Subscription, the changes made to which previously, are to be rolled back.
```
Comment (asimks) [inline Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1:24]: We get it from backed up details right?

File: Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1 [hunk 4]
```diff
@@ -0,0 +1,795 @@
...
+# Instructions to execute the script:
+    To remediate:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to configure WAF Policy on all endpoints of Front Door CDNs in a Subscription. Refer `Examples`, below.
+
+    To roll back:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to remove the configured WAF Policy on all endpoints of Front Door CDNs in a Subscription. Refer `Examples`, below.
```
Comment (asimks) [inline Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1:36]: We should say roll back the configuration to previous state as we dont remove it for all the CDNs in subscription.

File: Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1 (hunk 2 above)
Comment (jaiswalricha) [reply Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1:23]: Yes

Comment (asimks) [reply Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1:23]: Could we rollback without executing remediation command. Just thinking of scenarios where rollback could be run in isolation.

File: Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1 [hunk 5]
```diff
@@ -0,0 +1,795 @@
...
+        if($totalfrontDoors -gt 0)
+        {
+            $frontDoors | ForEach-Object {
+                $frontDoor = $_
+                $frontDoorId = $_.Id
+                $resourceGroupName = $_.ResourceGroupName
+                $frontDoorName = $_.Name
+
+                # Get all Frontendpoint(s) for this Front Door.
+                $frontendpoints = ( Get-AzFrontDoorCdnEndpoint -ResourceGroupName $resourceGroupName -ProfileName $frontDoorName -ErrorAction SilentlyContinue)
```
Comment (asimks) [inline Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1:253]: If there is failure in these commands, would we be able to identify difference between no endpoints and failure case?

File: Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1 (hunk 3 above)
Comment (jaiswalricha) [reply Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1:24]: Yes, after successfully remediation, we store the remediated End points and here we are explaining about rolling back the same.

File: Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1 (hunk 2 above)
Comment (jaiswalricha) [reply Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1:23]: we do roll back only for remediated resource only.

File: Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1 (hunk 4 above)
Comment (jaiswalricha) [reply Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1:36]: this is done

File: Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1 (hunk 5 above)
Comment (jaiswalricha) [reply Scripts/RemediationScripts/Remediate-ConfigureWAFOnAzureFrontDoorCDN.ps1:253]: I have included this change thru if else for now.

**Pull request merged**